File: string to event bug/server/src/main/java/bgu/spl/net/impl/Frames/ConnectedFrame.java
package bgu.spl.net.impl.Frames;

import java.util.Map;

import bgu.spl.net.impl.User.User;
import bgu.spl.net.impl.gamesUpdate.GamesFeed;
import bgu.spl.net.impl.stomp.StompCommand;
import bgu.spl.net.impl.stomp.StompFrame;

public class ConnectedFrame extends StompFrame{
    String version;

    public ConnectedFrame(Map<String,String> headers) {
        super(StompCommand.CONNECTED, headers);
    }

    @Override 
    public StompFrame create(String[] body){
        if(!stompsHeaders.containsKey("version")){
            missingHeaders.add("version");
            errorAccourd = true;
        }
        if(stompsHeaders.containsKey("receipt")){
            receiptId = stompsHeaders.get("receipt");
            errorHeaders.put("receipt", receiptId);
       }
       if(errorAccourd){
            return frameUtil.buildHeaderMissingErrorFrame(errorHeaders,missingHeaders);
       }

        version = stompsHeaders.get("version");
        return this;
    }


    @Override
    public void excute(GamesFeed gamesFeed, User user){
        gamesFeed.connections.send(user.getConnectionId(), this);
    }
}

File: string to event bug/client/src/Frames/ConnectedFrame.cpp
#include "../include/StompFrame.h"
#include <iostream>
using std::cout;

//we get it from the server
ConnectedFrame::ConnectedFrame(): version(""){
    command = CONNECTED;
}

ConnectedFrame::ConnectedFrame(string _version): version(_version){
    command = CONNECTED;
}

ConnectedFrame::~ConnectedFrame(){}

void ConnectedFrame::excute(user* curUser){
    curUser->setIsConnected();
}

void ConnectedFrame::create(map<string,string> headers, string body){
    if (headers.find("version") == headers.end()) {
        cout << "error - version not found"; // todo: remove
    }
    version = headers["version"];
}
